### DISCUSSION TITLE
Modal triggered by button inside dropdown list does not show up

### DISCUSSION THREAD

**Discussion by pedrovgp** [epicmaxco/vuestic-ui#3292] [Q&A] [answered]
Heelo!

What I am trying to do is:
1. Have a menu dropdown in a page (let's call it details page)
2. The dropdown has a list of buttons
3. Each button triggers a modal with a form

I have a Modal component defined. If I insert it directly in details page, it works as expected, but when I insert it into the dropdown list in the details page, it never shows up when its button is clicked.

Sample code:

```vue
// details page
<template>
  <death-modal :animal-id="props.animalId" />
  <animal-dropdown :animal-id="animal?.id" />
</template>


<script setup lang="ts">
  import AnimalDropdown from './registry/AnimalDropdown.vue'
  import Modal from './registry/Modal.vue'

</script>
```

``` vue
// animal-dropdown
<template>
  <va-dropdown>
    <template #anchor>
      <va-button icon="more_vert"> Registrar </va-button>
    </template>
    <va-dropdown-content>
      <va-list :fit="true">
        <va-list-item class="list__item">
          <modal :animal-id="props.animalId" />
        </va-list-item>
      </va-list>
    </va-dropdown-content>
  </va-dropdown>
</template>

<script setup lang="ts">
  import Modal from './Modal.vue'

  const props = defineProps<{
    animalId?: string | number
  }>()

</script>
```

```vue
// Modal
<template>
  <va-button @click="showContent = !showContent"> Registry </va-button>
    <va-modal v-model="showContent" :title="'Registry'" :message="'Test message'">
    </va-modal>
</template>

<script setup lang="ts">
  import { ref } from 'vue'
  const showContent = ref(false)

</script>
```

**Comment by rustem-nasyrov** [comment, marked as answer]
Hello, @pedrovgp! I found a solution for you. You need to add `.stop` [event modifier](https://vuejs.org/guide/essentials/event-handling.html#event-modifiers) to the `@click` listener. It will prevent bubbling event up. Also, I recommend you to wrap your `Modal.vue` template contents with `div`, to avoid vue warnings in the console.

```html
<template>
  <div>
    <va-button @click.stop="showContent = !showContent"> Registry </va-button>
    <va-modal v-model="showContent" :title="'Registry'" :message="'Test message'" />
  </div>
</template>

<script setup lang="ts">
  import { ref } from 'vue'
  const showContent = ref(false)
</script>
```

**Reply by pedrovgp** [reply]
Hi @rustem-nasyrov , thanks for the reply! It did work, but I actually have a form inside that modal and when I open it from within the dropdown, when I click in any of the fields, it dismisses the modal :(
